core/src/commonMain/kotlin/Observers.kt

```diff
@@ -81,7 +83,10 @@ internal class Observers<T>(
             .mapNotNull { event -> (event as? CharacteristicChange)?.data }
             .onCompletion {
                 try {
-                    observation.onCompletion(onSubscription)
+                    // If this suspending function is canceled the peripheral could end up in an indeterminate state.
```

Comment:
nitpick

```suggestion
                    // `NonCancellable` used to prevent interruption of resetting the observation
                    // state, which can prevent subsequent re-observation.
                    // See https://github.com/JuulLabs/kable/issues/677 for more details.
```